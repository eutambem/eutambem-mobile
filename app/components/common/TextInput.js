import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextInput as NativeTextInput } from 'react-native';
import appStyles from '../../styles';
import { MAX_TEXT_INPUT_LENGTH } from '../../constants';

class TextInput extends Component {
  blur = () => this.input.blur();

  inputRef = (component) => { this.input = component; }

  render() {
    let style = [appStyles.input];
    let autoGrow = false;
    if (this.props.multiline) {
      style = style.concat(appStyles.multilineInput);
      autoGrow = true;
    }

    return (
      <NativeTextInput
        style={style}
        maxLength={MAX_TEXT_INPUT_LENGTH}
        multiline={this.props.multiline}
        autoGrow={autoGrow}
        ref={this.inputRef}
        {...this.props}
      />
    );
  }
}

TextInput.propTypes = {
  multiline: PropTypes.bool,
};

TextInput.defaultProps = {
  multiline: false,
};

export default TextInput;
