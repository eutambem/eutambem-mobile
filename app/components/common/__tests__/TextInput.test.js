import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import TextInput from '../TextInput';

const props = {
  placeholder: 'Test TextInput',
};

describe('TextInput', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TextInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
