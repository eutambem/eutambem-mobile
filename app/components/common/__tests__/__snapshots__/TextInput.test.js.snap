// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  autoGrow={false}
  maxLength={350}
  multiline={false}
  placeholder="Test TextInput (Opcional)"
  required={false}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "borderBottomColor": "#91508c",
        "borderBottomWidth": 1,
        "borderWidth": 0,
        "height": 30,
        "margin": 0,
        "marginBottom": 10,
        "marginTop": 10,
      },
    ]
  }
/>
`;
